src/include/txn.i

```diff
@@ -538,34 +543,77 @@ __wt_txn_visible(
 #endif
 }
 
+/*
+ * __wt_txn_upd_visible_type --
+ *      Visible type of given update for the current transaction.
+ */
+static inline WT_VISIBLE_TYPE
+__wt_txn_upd_visible_type(WT_SESSION_IMPL *session, WT_UPDATE *upd)
+{
+	uint8_t upd_state;
+	bool upd_visible;
+
+	for (;;) {
+		if ((upd_state = upd->state) == WT_UPDATE_STATE_LOCKED) {
+			/* Commit is in progress, yield and try again. */
+			__wt_yield();
+			continue;
+		}
+
+		upd_visible = __wt_txn_visible(
+		    session, upd->txnid, WT_TIMESTAMP_NULL(&upd->timestamp));
+
+		/*
+		 * The visibility check is only valid if the update does not
+		 * change state.  If the state does change, recheck visibility.
+		 */
+		if (upd->state != upd_state)
+			continue;
```

Comment:
It seems odd we yield in the commit case, but not in the `update-changing` case. Why are they different?